saleor/graphql/discount/mutations/promotion_create.py

```diff
@@ -0,0 +1,224 @@
+                        )
+                    )
+                if "reward_value" not in rule_data:
+                    errors["reward_value"].append(
+                        ValidationError(
+                            "The rewardValue is required for when cataloguePredicate "
+                            "is provided.",
+                            code=PromotionCreateErrorCode.REQUIRED.value,
+                            params={"index": index},
+                        )
+                    )
+                rule_data["catalogue_predicate"] = clean_predicate(
+                    rule_data.get("catalogue_predicate")
+                )
+            cleaned_rules.append(rule_data)
+
+        return cleaned_rules, errors
+
+    @classmethod
+    def clean_channels(
+        cls,
+        info: ResolveInfo,
+        channel_ids: List[str],
+        index: int,
+        errors: DefaultDict[str, List[ValidationError]],
+    ) -> List[channel_models.Channel]:
+        try:
+            channels = get_nodes(channel_ids, Channel, schema=info.schema)
+        except GraphQLError as e:
+            errors["channels"].append(
+                ValidationError(
+                    str(e),
+                    code=PromotionCreateErrorCode.GRAPHQL_ERROR.value,
+                    params={"index": index},
+                )
+            )
+            return []
+        return channels
+
+    @classmethod
+    def perform_mutation(cls, _root, info: ResolveInfo, /, **data):
+        instance = cls.get_instance(info, **data)
+        data = data["input"]
+        cleaned_input = cls.clean_input(info, instance, data)
+        instance = cls.construct_instance(instance, cleaned_input)
+        manager = get_plugin_manager_promise(info.context).get()
+
+        cls.clean_instance(info, instance)
+        with traced_atomic_transaction():
+            cls.save(info, instance, cleaned_input)
+            cls._save_m2m(info, instance, cleaned_input)
+            cls.send_sale_notifications(manager, instance)
+        return cls.success_response(instance)
+
+    @classmethod
+    def _save_m2m(
+        cls, info: ResolveInfo, instance: models.Promotion, cleaned_data: dict
+    ):
+        super()._save_m2m(info, instance, cleaned_data)
+        rules_with_channels_to_add = []
+        if rules_data := cleaned_data.get("rules"):
+            rules = []
+            for rule_data in rules_data:
+                channels = rule_data.pop("channels", None)
+                rule = models.PromotionRule(promotion=instance, **rule_data)
+                rules_with_channels_to_add.append((rule, channels))
+                rules.append(rule)
+            models.PromotionRule.objects.bulk_create(rules)
+
+        for rule, channels in rules_with_channels_to_add:
+            rule.channels.set(channels)
+
+    @classmethod
+    def send_sale_notifications(cls, manager, instance):
+        # TODO: implement the notifications
```

Comment:
Will this issue be addressed in a separate pull request?

Reply:
Yep. I write about it in the PR description and it's on the `TODO` list on the RFC